modules/function/src/main/java/com/opengamma/strata/function/calculation/fx/FxNdfMultCurrencyAmountFunction.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * <p>
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.function.calculation.fx;
+
+import static com.opengamma.strata.calc.runner.function.FunctionUtils.toMultiCurrencyArray;
+
+import java.util.stream.IntStream;
+
+import com.opengamma.strata.basics.currency.MultiCurrencyAmount;
+import com.opengamma.strata.calc.marketdata.CalculationMarketData;
+import com.opengamma.strata.calc.runner.DefaultSingleCalculationMarketData;
+import com.opengamma.strata.calc.runner.function.result.ScenarioResult;
+import com.opengamma.strata.function.marketdata.MarketDataRatesProvider;
+import com.opengamma.strata.product.fx.ExpandedFxNdf;
+import com.opengamma.strata.product.fx.FxNdfTrade;
+
+/**
+ * Function which calculates values of a measure for an {@link FxNdfTrade} and returns a {@link MultiCurrencyAmount}.
+ */
+public abstract class FxNdfMultCurrencyAmountFunction extends AbstractFxNdfFunction<MultiCurrencyAmount> {
```

Comment:
Class name is wrong. It also has a different naming order to `MultiCurrencyAmountFxSingleFunction`, not sure which way around works best.